EIPS/eip-1185.md

```diff
@@ -0,0 +1,107 @@
+---
+eip: 1185
+title: Storage of DNS Records in ENS
+author: Jim McDonald <Jim@mcdee.net>
+status: Draft
+type: Standards Track
+category: ERC
+created: 2018-06-26
+requires: 137
+---
+  
+## Abstract
+This EIP defines a resolver profile for ENS that provides features for storage and lookup of DNS records. This allows ENS to be used as a store of authoritative DNS information.
+
+## Motivation
+ENS is a perfect store for DNS records.  It provides the distributed authority of DNS without conflating ownership and serving of information.  With ENS, the owner of a domain has full control over their own DNS records.  Separately, ENS has the ability (through smart contracts) for a domain's subdomains to be irrevocably assigned to another entity.
+
+## Specification
+
+The resolver profile to support DNS on ENS builds on the standalone resolver as defined in #137.
+
+Traditionally, DNS is a zone-based system in that all of the records for a zone are kept together in the same file.  This has the benefit of simplicity and atomicity of zone updates, but when transposed to ENS can result in significant gas costs for simple changes.  As a result, the resolver provides the ability to store either zone-based or record-based DNS information for a given domain, each with their own set of functions for managing the information.
+
+The primary reason that a user might choose to use the zone-based functions is if the zone uses information that spans multiple records and requires strict ordering (for example DNSSEC with NSSEC/NSSEC3 records), so an unordered record-based storage system is not possible.  If this is not a consideration then record-based storage is generally cheaper and more flexible.
+
+A DNS zone is uniquely identified by the domain for which it contains records.  A DNS record set is uniquely identified by the domain, label and resource record type.
+
+One of zone-based or record-based functions should be chosen for a given ENS domain.  If both are used then results from a DNS resolver may be inconsistent.
+
+The DNS resolver interface consists of three functions to update DNS information and three functions to query DNS information.
+
+### setDNSRecords(bytes32 node, bytes data)
```

Comment:
I'm concerned that the way record based storage is configured here requires the resolver to unpack wire format records and return them individually. Apart from the general complexity of that, doing it properly would require the resolver to implement punycode and potentially UTS46 normalisation - a big ask!

Reply:
The node is the namehash, so no problems there.  The wire format records can contain high codepage characters in their names or values, but that shouldn't be an issue as the records are not parsed beyond simple character counting.